tests/api_tests/applications/test_views.py

```diff
@@ -128,6 +128,44 @@ def setUp(self):
         self.user1_app = ApiOAuth2ApplicationFactory(owner=self.user1)
         self.user1_app_url = _get_application_detail_route(self.user1_app)
 
+        self.missing_id = {
+            'type': 'applications',
+            'name': 'A shiny new application',
+            'home_url': 'http://osf.io',
+            'callback_url': 'https://cos.io'
+        }
+
+        self.missing_type = {
+            'id': self.user1_app._id,
+            'name': 'A shiny new application',
+            'home_url': 'http://osf.io',
+            'callback_url': 'https://cos.io'
+        }
+
+        self.incorrect_id = {
+            'id': '12345',
+            'type': 'applications',
+            'name': 'A shiny new application',
+            'home_url': 'http://osf.io',
+            'callback_url': 'https://cos.io'
+        }
+
+        self.incorrect_type = {
+            'id': self.user1_app._id,
+            'type': 'Wrong type.',
+            'name': 'A shiny new application',
+            'home_url': 'http://osf.io',
+            'callback_url': 'https://cos.io'
+        }
+
+        self.correct =  {
+            'id': self.user1_app._id,
+            'type': 'applications',
+            'name': 'A shiny new application',
```

Comment:
Ack. I was so focused on adding `id` and `type` I didn't think about nesting everything else under `attributes` in the request.